source/app/service-providers/citeproc/extract-bibtex-attachments.ts

```diff
@@ -36,6 +36,18 @@ export default function extractBibtexAttachments (
   // Return value will be a fast-access dictionary
   let files = Object.create(null)
 
+  // First we search for the jabref comments containing the files' root directories
+  for (let entry of ast.children) {
+    if (entry.kind !== 'BracedComment') {
+      continue
+    }
+
+    // The format of the value field is 'jabref-meta: fileDirectory*:<path>;'
+    if (entry.value.split(':')[1].trim().startsWith('fileDirectory')) {
```

Comment:
Do all JabRef-comments start with `jabref-meta`? In that case, you can already reduce the complexity of the for loop by filtering:

```js
// Only retain comments
const comments = ast.children.filter(entry => entry.kind === 'BracedComment')
const jabrefComments = comments.filter(entry => entry.value.startsWith('jabref-meta:')

for (const entry of jabrefComments) {
  // ... your code
}
```